src/creator.rs

```diff
@@ -116,14 +114,17 @@ impl<H: Hasher> Creator<H> {
 
     pub(crate) async fn create(&mut self, mut exit: oneshot::Receiver<()>) {
         let half_hour = Duration::from_secs(30 * 60);
-        let mut round: usize = 0;
+        let mut round: Round = 0;
         let mut delay_fut = Delay::new((self.create_lag)(0)).fuse();
         let mut delay_passed = false;
         loop {
             futures::select! {
                 unit = self.parents_rx.next() => {
                     if let Some(u) = unit{
                         self.add_unit(u.round(), u.creator(), u.hash());
+                        if self.current_round < u.round() {
```

Comment:
I'm pretty sure this should happen in `init_round` rather than here, especially since I believe you might have introduced a bug where when we create a unit of a higher round than `current_round` we don't actually update `current_round`.